roles/matrix-synapse/templates/ext/appservice-irc/systemd/matrix-appservice-irc.service.j2

```diff
@@ -0,0 +1,26 @@
+[Unit]
+Description=Matrix Appservice IRC server
+After=docker.service
+Requires=docker.service
+Requires=matrix-synapse.service
+After=matrix-synapse.service
+
+[Service]
+Type=simple
+ExecStartPre=-/usr/bin/docker kill matrix-appservice-irc
+ExecStartPre=-/usr/bin/docker rm matrix-appservice-irc
+ExecStart=/usr/bin/docker run --rm --name matrix-appservice-irc \
+			--log-driver=none \
+			-e "UID={{ matrix_user_uid }}" -e "GID={{ matrix_user_gid }}" \
+			--network={{ matrix_docker_network }} \
+			-p 127.0.0.1:9999:9999 \
```

Comment:
Do we really need to expose port `9999` to the local host?

You can make this conditional (off by default) by doing a few things.

First, introducing a new variable in `roles/matrix-synapse/defaults/main.yml` called something like `matrix_app_service_container_expose_port` (default value of `false`). See `matrix_synapse_container_expose_client_server_api_port` at the beginning of that same defaults file for reference.

Then, wire this variable to whether matrix-nginx-proxy is enabled or not. We do this in the glue file (`group_vars/matrix-servers`).

```
matrix_app_service_container_expose_port: "{{ not matrix_nginx_proxy_enabled }}"
```

And finally, you can edit the `matrix-appservice-irc.service.j2` file to make it use it:

```

			{% if matrix_app_service_container_expose_port %}
			-p 127.0.0.1:9999:9999 \
			{% endif %}
```